src/target/riscv32.c

```diff
@@ -141,6 +145,41 @@ static void riscv32_regs_write(target_s *const target, const void *const data)
 	riscv_csr_write(hart, RV_DPC, &regs->pc);
 }
 
+static inline ssize_t riscv32_bool_to_4(const bool ret)
+{
+	return ret ? 4 : -1;
+}
+
+static ssize_t riscv32_reg_read(target_s *target, const int reg, void *data, const size_t max)
+{
+	if (max < 4)
```

Comment:
This still checks `max < 4` while the write checks `max != 4` - which is still inconsistent. Is there a good way to be more consistent about this?